lib/go/thrift/debug_protocol.go

```diff
@@ -25,9 +25,27 @@ import (
 )
 
 type TDebugProtocol struct {
-	Delegate  TProtocol
-	LogPrefix string
+	// Required. The actual TProtocol to do the read/write.
+	Delegate TProtocol
+
+	// Optional. The logger and prefix to log all the args/return values
+	// from Delegate TProtocol calls.
+	//
+	// If Logger is nil, StdLogger using stdlib log package with os.Stderr
+	// will be used. If disable logging is desired, set Logger to NopLogger
+	// explicitly instead of leaving it as nil/unset.
 	Logger    Logger
+	LogPrefix string
+
+	// Optional. An TProtocol to dump everything read/wrote from Delegate.
+	//
+	// A typical use case of this is to use TSimpleJSONProtocol wrapping
+	// TMemoryBuffer in a middleware to json logging requests/responses.
+	//
+	// This feature is not available from TDebugProtocolFactory. In order to
+	// use it you have to construct TDebugProtocol directly, or set Dump
+	// field after getting a TDebugProtocol from the factory.
+	Dump TProtocol
```

Comment:
I guess "Dump" is not the best name, and I'm open to suggestions :)

Reply:
Hmm, "Export"? "Intercept"?

Reply:
Those names don't sound right to me either 😕 What do you think about "WriteTo", "Writer", "DuplicateTo"?

Reply:
Of those, DuplicateTo sounds best :)

Reply:
Of those, DuplicateTo sounds best :)

Reply:
OK renamed to `DuplicateTo`. commit message also updated accordingly.